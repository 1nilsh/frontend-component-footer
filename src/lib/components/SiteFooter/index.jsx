import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-i18n';
import { Hyperlink } from '@edx/paragon';
import qs from 'query-string';

import messages from './SiteFooter.messages';

const EVENT_NAMES = {
  FOOTER_LINK: 'edx.bi.footer.link',
};


class SiteFooter extends React.Component {
  constructor(props) {
    super(props);
    this.externalLinkClickHandler = this.externalLinkClickHandler.bind(this);
    this.applyLanguageSelection = this.applyLanguageSelection.bind(this);
  }

  applyLanguageSelection(event) {
    event.preventDefault();
    const languageCode = event.target.elements['site-footer-language-select'].value;
    const { languageForm: { onLanguageSelected } } = this.props;
    onLanguageSelected(languageCode);
  }

  externalLinkClickHandler(event) {
    const label = event.currentTarget.getAttribute('href');
    const eventName = EVENT_NAMES.FOOTER_LINK;
    const properties = {
      category: 'outbound_link',
      label,
    };
    this.props.handleAllTrackEvents(eventName, properties);
  }

  renderSiteLogo() {
    return (
      <img
        src={this.props.siteLogo}
        alt={this.props.intl.formatMessage(
          messages['footer.site-footer.site-logo.alt-text'],
          { siteName: this.props.siteName },
        )}
      />
    );
  }

  renderMarketingSiteUrl(path) {
    return `${this.props.marketingSiteBaseUrl}${path}`;
  }

  renderEnterpriseMarketingSiteUrl(enterpriseLinkData) {
    const {
      queryParams,
      url,
    } = enterpriseLinkData;
    return queryParams ? `${url}/?${qs.stringify(queryParams)}` : url;
  }

  renderMobileLinks() {
    const {
      intl,
      siteName,
      showMobileLinks,
      appleAppStoreUrl,
      googlePlayUrl,
    } = this.props;
    let mobileLinks = null;
    if (showMobileLinks) {
      mobileLinks = (
        <ul className="d-flex flex-row justify-content-between list-unstyled max-width-264 p-0 mb-5">
          <li>
            <a href={appleAppStoreUrl} rel="noopener noreferrer" target="_blank" onClick={this.externalLinkClickHandler}>
              <img
                className="max-height-39"
                alt={intl.formatMessage(
                  messages['footer.site-footer.apple-app-store.alt-text'],
                  { siteName },
                )}
                src="https://prod-edxapp.edx-cdn.org/static/images/app/app_store_badge_135x40.d0558d910630.svg"
              />
            </a>
          </li>
          <li>
            <a href={googlePlayUrl} rel="noopener noreferrer" target="_blank" onClick={this.externalLinkClickHandler}>
              <img
                className="max-height-39"
                alt={intl.formatMessage(
                  messages['footer.site-footer.google-play.alt-text'],
                  { siteName },
                )}
                src="https://prod-edxapp.edx-cdn.org/static/images/app/google_play_badge_45.6ea466e328da.png"
              />
            </a>
          </li>
        </ul>
      );
    }
    return mobileLinks;
  }

  renderBusinessMarketingListItem() {
    const { siteName, enterpriseMarketingLink } = this.props;
    return enterpriseMarketingLink && (
      <li>
        <a href={this.renderEnterpriseMarketingSiteUrl(enterpriseMarketingLink)}>
          <FormattedMessage
            id="footer.site-footer.link.business"
            defaultMessage="{siteName} for Business"
            description="A link that points to a business marketing page for a specified website"
            values={{ siteName }}
          />
        </a>
      </li>
    );
  }

  render() {
    const {
      intl,
      siteName,
      openSourceUrl,
      termsOfServiceUrl,
      privacyPolicyUrl,
      contactUrl,
      supportUrl,
      socialLinks,
      supportedLanguages,
      languageForm,
    } = this.props;
    const showLanguageSelector = supportedLanguages.length > 0 &&
                                 languageForm;
    return (
      <footer
        role="contentinfo"
        aria-label={intl.formatMessage(messages['footer.site-footer.footer.aria-label'])}
        className="footer d-flex justify-content-center border-top py-3 px-4"
      >
        <div className="max-width-1180 d-grid">
          <div className="area-1">
            <Hyperlink
              destination={this.renderMarketingSiteUrl('/')}
              content={this.renderSiteLogo()}
              aria-label={intl.formatMessage(
                messages['footer.site-footer.site-logo.aria-label'],
                { siteName },
              )}
            />
            {showLanguageSelector &&
              <div className="i18n d-flex mt-2">
                <form
                  className="d-flex align-items-start"
                  onSubmit={this.applyLanguageSelection}
                >
                  {/* eslint-disable-next-line jsx-a11y/label-has-for */}
                  <label htmlFor="site-footer-language-select">
                    {languageForm.icon}
                    <div className="sr-only">{languageForm.screenReaderLabel}</div>
                  </label>
                  <select
                    id="site-footer-language-select"
                    className="mx-2 mt-1"
                    name="site-footer-language-select"
                    defaultValue={languageForm.activeLanguage}
                  >
                    {supportedLanguages.map(({ value, label }) =>
                      <option key={value} value={value}>{label}</option>)}
                  </select>
                  <button className="mt-1" type="submit">{languageForm.submitLabel}</button>
                </form>
              </div>
            }
          </div>
          <div className="area-2">
            <h2>{siteName}</h2>
            <ul className="list-unstyled p-0 m-0">
              <li>
                <a href={this.renderMarketingSiteUrl('/about-us')}>
                  <FormattedMessage
                    id="footer.site-footer.link.about"
                    defaultMessage="About"
                  />
                </a>
              </li>
              {this.renderBusinessMarketingListItem()}
              <li>
                <a href={this.renderMarketingSiteUrl('/affiliate-program')}>
                  <FormattedMessage
                    id="footer.site-footer.link.affiliates"
                    defaultMessage="Affiliates"
                  />
                </a>
              </li>
              <li>
                <a href={openSourceUrl}>
                  <FormattedMessage
                    id="footer.site-footer.link.open-source"
                    defaultMessage="Open {siteName}"
                    values={{ siteName }}
                    description="Open Source link text"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/careers')}>
                  <FormattedMessage
                    id="footer.site-footer.link.careers"
                    defaultMessage="Careers"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/news-announcements')}>
                  <FormattedMessage
                    id="footer.site-footer.link.news"
                    defaultMessage="News"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div className="area-3">
            <h2>
              <FormattedMessage
                id="footer.site-footer.link.header.legal"
                defaultMessage="Legal"
                description="Header for legal links"
              />
            </h2>
            <ul className="list-unstyled p-0 m-0">
              <li>
                <a href={termsOfServiceUrl}>
                  <FormattedMessage
                    id="footer.site-footer.link.terms-of-service"
                    defaultMessage="Terms of Service & Honor Code"
                  />
                </a>
              </li>
              <li>
                <a href={privacyPolicyUrl}>
                  <FormattedMessage
                    id="footer.site-footer.link.privacy"
                    defaultMessage="Privacy Policy"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/accessibility')}>
                  <FormattedMessage
                    id="footer.site-footer.link.accessibility"
                    defaultMessage="Accessibility Policy"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/trademarks')}>
                  <FormattedMessage
                    id="footer.site-footer.link.trademark"
                    defaultMessage="Trademark Policy"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/sitemap')}>
                  <FormattedMessage
                    id="footer.site-footer.link.sitemap"
                    defaultMessage="Sitemap"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div className="area-4">
            <h2>
              <FormattedMessage
                id="footer.site-footer.link.header.connect"
                defaultMessage="Connect"
                description="Header for connect links"
              />
            </h2>
            <ul className="list-unstyled p-0 m-0">
              <li>
                <a href={this.renderMarketingSiteUrl('/blog')}>
                  <FormattedMessage
                    id="footer.site-footer.link.blog"
                    defaultMessage="Blog"
                  />
                </a>
              </li>
              <li>
                <a href={contactUrl}>
                  <FormattedMessage
                    id="footer.site-footer.link.contact-us"
                    defaultMessage="Contact Us"
                  />
                </a>
              </li>
              <li>
                <a href={supportUrl}>
                  <FormattedMessage
                    id="footer.site-footer.link.help-center"
                    defaultMessage="Help Center"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/media-kit')}>
                  <FormattedMessage
                    id="footer.site-footer.link.media-kit"
                    defaultMessage="Media Kit"
                  />
                </a>
              </li>
              <li>
                <a href={this.renderMarketingSiteUrl('/donate')}>
                  <FormattedMessage
                    id="footer.site-footer.link.donate"
                    defaultMessage="Donate"
                  />
                </a>
              </li>
            </ul>
          </div>
          <div className="area-5">
            {socialLinks.length > 0 &&
              <ul className="d-flex flex-row justify-content-between list-unstyled max-width-222 p-0 mb-4">
                {socialLinks.map(link => (
                  <li key={link.url}>
                    <a
                      href={link.url}
                      title={link.title}
                      rel="noopener noreferrer"
                      target="_blank"
                      onClick={this.externalLinkClickHandler}
                    >
                      {link.icon}
                      <span className="sr-only">{link.screenReaderText}</span>
                    </a>
                  </li>
                ))}
              </ul>
            }
            {this.renderMobileLinks()}
            <p>
              <FormattedMessage
                id="footer.site-footer.copyright-text"
                defaultMessage="{copyrightSymbol} {startDate}–{endDate} {siteName} Inc."
                values={{
                  copyrightSymbol: '©',
                  startDate: '2012',
                  endDate: `${new Date().getFullYear()}`,
                  siteName,
                }}
                description="Footer copyright text with copyright symbol and dates"
              />
              <br />
              <FormattedMessage
                id="footer.site-footer.trademark-text"
                defaultMessage="EdX, Open edX, and MicroMasters are registered trademarks of edX Inc. | 深圳市恒宇博科技有限公司 {icpLicense}"
                values={{ icpLicense: <a href="http://www.beian.miit.gov.cn">粤ICP备17044299号-2</a> }}
                description="Footer trademark text"
              />
            </p>
          </div>
        </div>
      </footer>
    );
  }
}

SiteFooter.propTypes = {
  siteName: PropTypes.string,
  siteLogo: PropTypes.node,
  marketingSiteBaseUrl: PropTypes.string,
  enterpriseMarketingLink: PropTypes.shape({
    url: PropTypes.string,
    queryParams: PropTypes.shape({}),
  }),
  supportUrl: PropTypes.string,
  contactUrl: PropTypes.string,
  openSourceUrl: PropTypes.string,
  termsOfServiceUrl: PropTypes.string,
  privacyPolicyUrl: PropTypes.string,
  socialLinks: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    icon: PropTypes.element.isRequired,
    screenReaderText: PropTypes.string.isRequired,
  })),
  showMobileLinks: PropTypes.bool,
  appleAppStoreUrl: PropTypes.string,
  googlePlayUrl: PropTypes.string,
  supportedLanguages: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })),
  languageForm: PropTypes.shape({
    screenReaderLabel: PropTypes.string.isRequired,
    submitLabel: PropTypes.string.isRequired,
    onLanguageSelected: PropTypes.func.isRequired,
    icon: PropTypes.node.isRequired,
    activeLanguage: PropTypes.string,
  }),
  handleAllTrackEvents: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
};

SiteFooter.defaultProps = {
  siteName: null,
  siteLogo: null,
  marketingSiteBaseUrl: null,
  enterpriseMarketingLink: null,
  supportUrl: null,
  contactUrl: null,
  openSourceUrl: null,
  termsOfServiceUrl: null,
  privacyPolicyUrl: null,
  socialLinks: [],
  showMobileLinks: true,
  appleAppStoreUrl: null,
  googlePlayUrl: null,
  supportedLanguages: [],
  languageForm: null,
};

export default injectIntl(SiteFooter);
export { EVENT_NAMES };
