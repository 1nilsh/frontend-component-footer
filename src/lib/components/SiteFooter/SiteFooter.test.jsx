import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { IntlProvider } from 'react-intl';

import FooterLogo from '../../../edx-footer.png';
import SiteFooter, { EVENT_NAMES } from './index';

const completeSiteFooterComponent = mockHandleAllTrackEvents =>
  (
    <IntlProvider locale="en">
      <SiteFooter
        siteName="example"
        siteLogo={FooterLogo}
        marketingSiteBaseUrl="https://www.example.com"
        supportUrl="https://www.example.com/support"
        contactUrl="https://www.example.com/contact"
        openSourceUrl="https://www.example.com/open"
        termsOfServiceUrl="https://www.example.com/terms-of-service"
        privacyPolicyUrl="https://www.example.com/privacy-policy"
        facebookUrl="https://www.facebook.com"
        twitterUrl="https://www.twitter.com"
        youTubeUrl="https://www.youtube.com"
        linkedInUrl="https://www.linkedin.com"
        redditUrl="https://reddit.com"
        googlePlusUrl="https://plus.google.com"
        appleAppStoreUrl="https://store.apple.com"
        googlePlayUrl="https://play.google.com"
        handleAllTrackEvents={mockHandleAllTrackEvents}
      />
    </IntlProvider>
  );

describe('<SiteFooter />', () => {
  describe('renders correctly', () => {
    it('renders with social and mobile links', () => {
      const mockHandleAllTrackEvents = jest.fn();
      const tree = renderer
        .create(completeSiteFooterComponent(mockHandleAllTrackEvents))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not render social links', () => {
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <SiteFooter
              siteName="example"
              siteLogo={FooterLogo}
              marketingSiteBaseUrl="https://www.example.com"
              supportUrl="https://www.example.com/support"
              contactUrl="https://www.example.com/contact"
              openSourceUrl="https://www.example.com/open"
              termsOfServiceUrl="https://www.example.com/terms-of-service"
              privacyPolicyUrl="https://www.example.com/privacy-policy"
              facebookUrl="https://www.facebook.com"
              twitterUrl="https://www.twitter.com"
              youTubeUrl="https://www.youtube.com"
              linkedInUrl="https://www.linkedin.com"
              redditUrl="https://reddit.com"
              googlePlusUrl="https://plus.google.com"
              appleAppStoreUrl="https://store.apple.com"
              googlePlayUrl="https://play.google.com"
              handleAllTrackEvents={jest.fn()}
              showSocialLinks={false}
            />
          </IntlProvider>
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not render mobile links', () => {
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <SiteFooter
              siteName="example"
              siteLogo={FooterLogo}
              marketingSiteBaseUrl="https://www.example.com"
              supportUrl="https://www.example.com/support"
              contactUrl="https://www.example.com/contact"
              openSourceUrl="https://www.example.com/open"
              termsOfServiceUrl="https://www.example.com/terms-of-service"
              privacyPolicyUrl="https://www.example.com/privacy-policy"
              facebookUrl="https://www.facebook.com"
              twitterUrl="https://www.twitter.com"
              youTubeUrl="https://www.youtube.com"
              linkedInUrl="https://www.linkedin.com"
              redditUrl="https://reddit.com"
              googlePlusUrl="https://plus.google.com"
              appleAppStoreUrl="https://store.apple.com"
              googlePlayUrl="https://play.google.com"
              handleAllTrackEvents={jest.fn()}
              showMobileLinks={false}
            />
          </IntlProvider>
        )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('handles analytics', () => {
    it('calls handleAllTrackEvents prop when external links clicked', () => {
      const mockHandleAllTrackEvents = jest.fn();
      const footer = mount((completeSiteFooterComponent(mockHandleAllTrackEvents)));
      const externalLinks = footer.find("a[target='_blank']");

      expect(externalLinks).toHaveLength(8);
      externalLinks.forEach((externalLink) => {
        const callIndex = mockHandleAllTrackEvents.mock.calls.length;
        externalLink.simulate('click');
        expect(mockHandleAllTrackEvents.mock.calls[callIndex]).toEqual([
          EVENT_NAMES.FOOTER_LINK,
          {
            category: 'outbound_link',
            label: externalLink.prop('href'),
          },
        ]);
      });
    });
  });
});
