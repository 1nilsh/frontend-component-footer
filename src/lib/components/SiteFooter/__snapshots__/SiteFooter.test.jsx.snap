// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteFooter /> does not render mobile links 1`] = `
<footer
  aria-label="Page Footer"
  className="footer d-flex justify-content-center border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="max-width-1180 d-grid"
  >
    <div
      className="area-1"
    >
      <a
        aria-label="example Home"
        href="https://www.example.com/"
        onClick={[Function]}
        target="_self"
      >
        <img
          alt="example logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="area-2"
    >
      <h2>
        example
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/about-us"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/enterprise"
          >
            example
             for Business
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/affiliate-program"
          >
            Affiliates
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/open"
          >
            Open 
            example
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/careers"
          >
            Careers
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/news-announcements"
          >
            News
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-3"
    >
      <h2>
        Legal
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/terms-of-service"
          >
            Terms of Service & Honor Code
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/privacy-policy"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/accessibility"
          >
            Accessibility Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/trademarks"
          >
            Trademark Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/sitemap"
          >
            Sitemap
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-4"
    >
      <h2>
        Connect
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/contact"
          >
            Contact Us
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/support"
          >
            Help Center
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/media-kit"
          >
            Media Kit
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/donate"
          >
            Donate
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-5"
    >
      <ul
        className="d-flex flex-row justify-content-between list-unstyled max-width-222 p-0 mb-4"
      >
        <li>
          <a
            href="https://www.facebook.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Facebook"
          >
            <span
              aria-hidden={true}
              className="fa fa-facebook-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Like example on Facebook
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.twitter.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter"
          >
            <span
              aria-hidden={true}
              className="fa fa-twitter-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on Twitter
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.youtube.com"
            rel="noopener noreferrer"
            target="_blank"
            title="Youtube"
          >
            <span
              aria-hidden={true}
              className="fa fa-youtube-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Subscribe to the example YouTube channel
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="LinkedIn"
          >
            <span
              aria-hidden={true}
              className="fa fa-linkedin-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on LinkedIn
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://plus.google.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Google+"
          >
            <span
              aria-hidden={true}
              className="fa fa-google-plus-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on Google+
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://reddit.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Reddit"
          >
            <span
              aria-hidden={true}
              className="fa fa-reddit-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Subscribe to the example subreddit
            </span>
          </a>
        </li>
      </ul>
      <p>
        © 2012–
        2019
         
        example
         Inc.
        <br />
        EdX, Open edX, and MicroMasters are registered trademarks of edX Inc. | 粤ICP备17044299号-2
      </p>
    </div>
  </div>
</footer>
`;

exports[`<SiteFooter /> does not render social links 1`] = `
<footer
  aria-label="Page Footer"
  className="footer d-flex justify-content-center border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="max-width-1180 d-grid"
  >
    <div
      className="area-1"
    >
      <a
        aria-label="example Home"
        href="https://www.example.com/"
        onClick={[Function]}
        target="_self"
      >
        <img
          alt="example logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="area-2"
    >
      <h2>
        example
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/about-us"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/enterprise"
          >
            example
             for Business
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/affiliate-program"
          >
            Affiliates
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/open"
          >
            Open 
            example
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/careers"
          >
            Careers
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/news-announcements"
          >
            News
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-3"
    >
      <h2>
        Legal
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/terms-of-service"
          >
            Terms of Service & Honor Code
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/privacy-policy"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/accessibility"
          >
            Accessibility Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/trademarks"
          >
            Trademark Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/sitemap"
          >
            Sitemap
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-4"
    >
      <h2>
        Connect
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/contact"
          >
            Contact Us
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/support"
          >
            Help Center
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/media-kit"
          >
            Media Kit
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/donate"
          >
            Donate
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-5"
    >
      <ul
        className="d-flex flex-row justify-content-between list-unstyled max-width-264 p-0 mb-5"
      >
        <li>
          <a
            href="https://store.apple.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download the example mobile app from the Apple App Store"
              className="max-height-39"
              src="https://prod-edxapp.edx-cdn.org/static/images/app/app_store_badge_135x40.d0558d910630.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="https://play.google.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download the example mobile app from Google Play"
              className="max-height-39"
              src="https://prod-edxapp.edx-cdn.org/static/images/app/google_play_badge_45.6ea466e328da.png"
            />
          </a>
        </li>
      </ul>
      <p>
        © 2012–
        2019
         
        example
         Inc.
        <br />
        EdX, Open edX, and MicroMasters are registered trademarks of edX Inc. | 粤ICP备17044299号-2
      </p>
    </div>
  </div>
</footer>
`;

exports[`<SiteFooter /> renders correctly 1`] = `
<footer
  aria-label="Page Footer"
  className="footer d-flex justify-content-center border-top py-3 px-4"
  role="contentinfo"
>
  <div
    className="max-width-1180 d-grid"
  >
    <div
      className="area-1"
    >
      <a
        aria-label="example Home"
        href="https://www.example.com/"
        onClick={[Function]}
        target="_self"
      >
        <img
          alt="example logo"
          src="test-file-stub"
        />
      </a>
    </div>
    <div
      className="area-2"
    >
      <h2>
        example
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/about-us"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/enterprise"
          >
            example
             for Business
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/affiliate-program"
          >
            Affiliates
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/open"
          >
            Open 
            example
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/careers"
          >
            Careers
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/news-announcements"
          >
            News
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-3"
    >
      <h2>
        Legal
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/terms-of-service"
          >
            Terms of Service & Honor Code
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/privacy-policy"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/accessibility"
          >
            Accessibility Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/trademarks"
          >
            Trademark Policy
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/sitemap"
          >
            Sitemap
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-4"
    >
      <h2>
        Connect
      </h2>
      <ul
        className="list-unstyled p-0 m-0"
      >
        <li>
          <a
            href="https://www.example.com/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/contact"
          >
            Contact Us
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/support"
          >
            Help Center
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/media-kit"
          >
            Media Kit
          </a>
        </li>
        <li>
          <a
            href="https://www.example.com/donate"
          >
            Donate
          </a>
        </li>
      </ul>
    </div>
    <div
      className="area-5"
    >
      <ul
        className="d-flex flex-row justify-content-between list-unstyled max-width-222 p-0 mb-4"
      >
        <li>
          <a
            href="https://www.facebook.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Facebook"
          >
            <span
              aria-hidden={true}
              className="fa fa-facebook-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Like example on Facebook
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.twitter.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter"
          >
            <span
              aria-hidden={true}
              className="fa fa-twitter-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on Twitter
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.youtube.com"
            rel="noopener noreferrer"
            target="_blank"
            title="Youtube"
          >
            <span
              aria-hidden={true}
              className="fa fa-youtube-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Subscribe to the example YouTube channel
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://www.linkedin.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="LinkedIn"
          >
            <span
              aria-hidden={true}
              className="fa fa-linkedin-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on LinkedIn
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://plus.google.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Google+"
          >
            <span
              aria-hidden={true}
              className="fa fa-google-plus-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Follow example on Google+
            </span>
          </a>
        </li>
        <li>
          <a
            href="https://reddit.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
            title="Reddit"
          >
            <span
              aria-hidden={true}
              className="fa fa-reddit-square fa-2x"
              id="Icon1"
            />
            <span
              className="sr-only"
            >
              Subscribe to the example subreddit
            </span>
          </a>
        </li>
      </ul>
      <ul
        className="d-flex flex-row justify-content-between list-unstyled max-width-264 p-0 mb-5"
      >
        <li>
          <a
            href="https://store.apple.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download the example mobile app from the Apple App Store"
              className="max-height-39"
              src="https://prod-edxapp.edx-cdn.org/static/images/app/app_store_badge_135x40.d0558d910630.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="https://play.google.com"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Download the example mobile app from Google Play"
              className="max-height-39"
              src="https://prod-edxapp.edx-cdn.org/static/images/app/google_play_badge_45.6ea466e328da.png"
            />
          </a>
        </li>
      </ul>
      <p>
        © 2012–
        2019
         
        example
         Inc.
        <br />
        EdX, Open edX, and MicroMasters are registered trademarks of edX Inc. | 粤ICP备17044299号-2
      </p>
    </div>
  </div>
</footer>
`;
